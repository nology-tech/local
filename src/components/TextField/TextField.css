@import url("https://fonts.googleapis.com/css2?family=Inter&family=Lato&family=Roboto&display=swap");
* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

/*                          Colours                         */
/*                          Font stuff              */
* {
  font-family: "Trebuchet MS", "Lucida Sans Unicode", "Lucida Grande", "Lucida Sans", Arial, sans-serif;
  color: #000000;
}

h1,
h2,
h3,
h4,
h5,
h6,
p {
  margin: 0.5rem 0;
  display: inline-block;
  width: 100%;
}

button {
  font-size: 1rem;
}

h1 {
  font-size: 2.1rem;
}

h2 {
  font-size: 1.65rem;
}

h3 {
  font-size: 1.25rem;
}

@media screen and (min-width: 768px) {
  :root {
    font-size: 110%;
  }
}
@media screen and (min-width: 1024px) {
  :root {
    font-size: 120%;
  }
}
.text-field {
  font-family: "Inter";
  width: 100%;
  display: flex;
  flex-flow: column wrap;
  align-items: center;
  padding: 0.4rem;
}
.text-field__label, .text-field__input {
  color: #000000;
  font-family: "Inter";
  font-weight: 400;
  font-size: 1rem;
  padding: 0.3rem;
}
.text-field__label {
  align-self: flex-start;
  padding-left: 0.7rem;
  padding-bottom: 0.7rem;
}
.text-field__input {
  height: 3.5rem;
  width: 100%;
  display: block;
  background-color: #f3f2e8;
  border-style: none;
  border: 0.125rem solid #1f1b1b;
  border-radius: 2rem;
  box-shadow: 0 0.25rem 0.25rem 0 rgba(0, 0, 0, 0.25);
}
.text-field__input--large {
  height: 7rem;
}/*# sourceMappingURL=TextField.css.map */