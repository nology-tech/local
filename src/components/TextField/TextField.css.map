{"version":3,"sources":["../../styles/_typography.scss","../../styles/_reset.scss","TextField.css","../../styles/_variables.scss","TextField.scss","../../styles/_mixins.scss"],"names":[],"mappings":"AACQ,oGAAA;ACDR;EACE,UAAA;EACA,SAAA;EACA,sBAAA;ACEF;;ACEA,6DAAA;AAuBA,qDAAA;AHxBA;EACE,qGAAA;EAEA,cGCW;ADEb;;AFAA;;;;;;;EAQE,gBAAA;EACA,qBAAA;EACA,WAAA;AEEF;;AFCA;EACE,eAAA;AEEF;;AFCA;EACE,iBAAA;AEEF;;AFCA;EACE,kBAAA;AEEF;;AFCA;EACE,kBAAA;AEEF;;AFCA;EAEE;IACE,eAAA;EECF;AACF;AFEA;EAEE;IACE,eAAA;EEDF;AACF;AE9CA;EACE,oBD6BiB;EC5BjB,WAAA;EACA,aAAA;EACA,sBAAA;EACA,mBAAA;EACA,eAAA;AFgDF;AE/CE;EAEE,cDJS;EEPX,oBFgCiB;EE/BjB,gBFoCoB;EEnCpB,eFuCkB;EC5BhB,eAAA;AFkDJ;AEhDE;EACE,sBAAA;EACA,oBAAA;EACA,sBAAA;AFkDJ;AE/CE;EACE,cAAA;EACA,WAAA;EACA,cAAA;EACA,yBDPI;ECQJ,kBAAA;EACA,8BAAA;EACA,mBAAA;EACA,mDAAA;AFiDJ;AEhDI;EACE,YAAA;AFkDN","file":"TextField.css"}