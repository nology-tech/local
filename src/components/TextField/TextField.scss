@use "../../../src/styles/reset" as *;
@use "../../../src/styles/variables" as *;
@use "../../styles/typography" as *;
@use "../../styles/mixins" as *;

.text-field {
  font-family: $text-font-family;
  width: 100%;
  display: flex;
  flex-flow: column wrap;
  align-items: center;
  padding: 0.4rem;

  &__label,
  &__input {
    color: $text-color;
    @include text-normal;
    padding: 0.3rem;
  }

  &__label {
    align-self: flex-start;
    padding-left: 0.7rem;
    padding-bottom: 0.7rem;
  }

  &__input {
    height: 3.5rem;
    width: 100%;
    display: block;
    background-color: $white;
    border-style: none;
    border: 0.125rem solid $black;
    border-radius: 2rem;
    box-shadow: 0 0.25rem 0.25rem 0 rgba(0, 0, 0, 0.25);

    &--large {
      height: 7rem;
    }
  }

  &__input--signin,
  &__input--signup {
    border: 0.1rem solid $black;
    border-radius: 0.625rem;
  }
}

.text-field--location-search-bar {
  display: block;
}

.text-field__label--location-search-bar {
  display: none;
}

.text-field__input--location-search-bar {
  width: 100%;
  text-align: center;
  display: flex;
  justify-content: space-around;
  align-content: space-between;

  &::placeholder {
    color: $text-color;
    text-align: center;
    font-weight: 500;
    opacity: 50%;
  }
}
