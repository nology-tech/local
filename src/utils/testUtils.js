import { jest } from "@jest/globals";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import UserContext from "../context/UserContext";
import mockUser from "../data/mockUser";

export const customRender = (
  ui,
  useRouting = true,
  contextOptions = {
    useContext: true,
    value: { user: mockUser, setUser: jest.fn() },
  }
) => {
  // wrap components in routing and context if requested
  const routingUiResult = useRouting && wrapWithRouting(ui);
  const contextUiResult =
    contextOptions.useContext &&
    wrapWithContext(routingUiResult, contextOptions.value);

  // use RTL's render function to return the test component
  return render(contextUiResult);
};

const wrapWithRouting = (ui) => {
  return <Router>{ui}</Router>;
};

const wrapWithContext = (ui, value) => {
  return <UserContext.Provider value={value}>{ui}</UserContext.Provider>;
};
