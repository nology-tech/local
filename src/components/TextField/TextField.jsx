import "./TextField.scss";

const TextField = ({
  uniqueId,
  inputType,
  labelHeader,
  modifier,
  onChange,
}) => {
  let inputClassName = "text-field__input";

  if (modifier) {
    inputClassName = `text-field__input text-field__input--${modifier}`;
  }

  let labelClassName = "text-field__label";

  if (modifier) {
    labelClassName = `text-field__label text-field__label--${modifier}`;
  }

  return (
    <div className="text-field">
      <label htmlFor={uniqueId} className={labelClassName}>
        {labelHeader}
      </label>
      <input
        id={uniqueId}
        type={inputType}
        className={inputClassName}
        onChange={onChange}
      />
    </div>
  );
};

export default TextField;
