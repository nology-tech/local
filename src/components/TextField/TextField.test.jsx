import TextField from "./TextField";
import { customRender } from "../../utils/testUtils";
import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

it("should render the TextField component", () => {
  const { container } = customRender(<TextField />);

  expect(container).toMatchSnapshot();
});

it("should call the function when user types in input box", () => {
  const mockFunction = jest.fn();
  customRender(
    <TextField onChange={mockFunction} labelHeader="jim" uniqueId="tracey" />
  );

  const textField = screen.getByLabelText("jim");

  userEvent.type(textField, "hi");

  expect(mockFunction).toBeCalled();
});
