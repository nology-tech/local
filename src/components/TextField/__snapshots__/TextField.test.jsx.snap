// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the TextField component 1`] = `
<div>
  <div
    class="text-field"
  >
    <label
      class="text-field__label"
    />
    <input
      class="text-field__input"
    />
  </div>
</div>
`;
